modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/exportimport/portlet/preferences/processor/DLExportImportPortletPreferencesProcessor.java

```diff
@@ -129,20 +149,27 @@ public PortletPreferences processExportPortletPreferences(
 			}
 		}
 
-		long selectedRepositoryId = GetterUtil.getLong(
-			portletPreferences.getValue("selectedRepositoryId", null));
-
-		if (!_exportImportHelper.isExportPortletData(portletDataContext) ||
-			(selectedRepositoryId != portletDataContext.getGroupId())) {
-
-			if (ExportImportThreadLocal.isStagingInProcess() &&
-				(selectedRepositoryId > 0)) {
+		if (!Validator.isBlank(selectedGroupExternalReferenceCode)) {
+			Group selectedGroup =
+				_groupLocalService.fetchGroupByExternalReferenceCode(
+					selectedGroupExternalReferenceCode,
+					portletDataContext.getCompanyId());
+
+			if (!_exportImportHelper.isExportPortletData(portletDataContext) ||
+				(selectedGroup.getGroupId() !=
+					portletDataContext.getGroupId())) {
+
+				if (ExportImportThreadLocal.isStagingInProcess()) {
+					_saveStagingPreferencesMapping(
+						portletDataContext, null,
+						selectedGroupExternalReferenceCode,
+						portletPreferences.getValue(
+							_PREFERENCE_KEY_SELECTED_REPOSITORY_EXTERNAL_REFERENCE_CODE,
+							null));
+				}
 
-				_saveStagingPreferencesMapping(
-					selectedRepositoryId, null, portletDataContext);
+				return portletPreferences;
 			}
-
-			return portletPreferences;
 		}
 
 		// Root folder ID is not set, we need to export everything
```

Comment:
Very minor thing, but we should update this comment to reference the ERC. Otherwise it may confuse readers.

Reply:
See change here: [cc09b54](https://github.com/liferay-content-management/liferay-portal/pull/6244/commits/cc09b548151fa6f38670c34e282c6619a87cf855)